src/atomate2/classical_md/openmm/flows/core.py

```diff
@@ -0,0 +1,159 @@
+"""Core flows for OpenMM module."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from jobflow import Flow
+
+from atomate2.classical_md.openmm.jobs.core import NVTMaker, TempChangeMaker
+from atomate2.classical_md.utils import create_list_summing_to
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from emmet.core.classical_md import ClassicalMDTaskDocument
+    from openff.interchange import Interchange
+
+    from atomate2.classical_md.openmm.jobs.base import BaseOpenMMMaker
+
+
+@dataclass
+class OpenMMFlowMaker:
+    """Run a production simulation.
+
+    This flexible flow links together any flows of OpenMM jobs.
+
+    Attributes
+    ----------
+    name : str
+        The name of the production job. Default is "production".
+    tags : list[str]
+        Tags to apply to the final job.
+    makers: list[BaseOpenMMMaker]
+        A list of makers to string together.
+    """
+
+    name: str = "flexible"
+    tags: list[str] = field(default_factory=list)
+    makers: list[BaseOpenMMMaker | OpenMMFlowMaker] = field(default_factory=list)
+
+    def make(
+        self,
+        interchange: Interchange | bytes,
+        prev_task: ClassicalMDTaskDocument | None = None,
+        output_dir: str | Path | None = None,
+    ) -> Flow:
+        """Run the production simulation using the provided Interchange object.
+
+        Parameters
+        ----------
+        interchange : Interchange
+            The Interchange object containing the system
+            to simulate.
+        prev_task : Optional[ClassicalMDTaskDocument]
+            The previous task to use as
+            a starting point for the production simulation.
+        output_dir : Optional[Union[str, Path]]
+            The directory to write reporter files to.
+
+        Returns
+        -------
+        Flow
+            A Flow object containing the OpenMM jobs for the simulation.
+        """
+        if len(self.makers) == 0:
+            raise ValueError("At least one maker must be included")
+
+        jobs: list = []
+        for i, maker in enumerate(self.makers):
+            # last job
+            if i == len(self.makers) - 1:
+                maker.tags = (maker.tags or []) + (self.tags or []) or None
+
+            job = maker.make(
+                interchange=interchange,
+                prev_task=prev_task,
+                output_dir=output_dir,
+            )
+            interchange = job.output.interchange
+            prev_task = job.output
+            jobs.append(job)
+
+        return Flow(
+            jobs,
+            output=jobs[-1].output,
+        )
+
+    @classmethod
+    def anneal_flow(
```

Comment:
Would it make sense to turn this into's its own class?

Reply:
Let me think about this. I agree that a monolithic flow maker isn't great design.